### PR TITLE
Dbus service for  an entitlement status

### PR DESCRIPTION
**Pull request by jstavel**
I have added a new object 'Entitlement'.
I have finished a proposal for `getPools` and 'getSubscriptions' methods in the new object.

### CODE REVIEW

File: docs/subscription-manager/dbus_objects.md [hunk 1]
```diff
@@ -94,3 +94,66 @@ creates.
   order is `organization, activation key list, options dictionary`.  The options
   dictionary can contain the same values as `Register`.  This call returns the
   JSON response from the subscription management server.
+
+# Entitlement
+* Bus name: `com.redhat.RHSM1`
+* Interfaces: `com.redhat.RHSM1.Environment`
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: Should be `com.redhat.RHSM1.Entitlement`

File: docs/subscription-manager/dbus_objects.md [hunk 2]
```diff
@@ -94,3 +94,66 @@ creates.
...
+* Bus path: `/com/redhat/RHSM1/Environment`
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: Should be `com/redhat/RHSM1/Entitlement`

File: docs/subscription-manager/dbus_objects.md [hunk 3]
```diff
@@ -94,3 +94,66 @@ creates.
...
+
+The Environment object offers an information about currently entitled environment.
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: I would rephrase as "The Entitlement object provides information about entitlements usable by a system"

File: docs/subscription-manager/dbus_objects.md [hunk 4]
```diff
@@ -94,3 +94,66 @@ creates.
...
+Entitled environment includes:
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: "Entitlement information includes:"

File: docs/subscription-manager/dbus_objects.md [hunk 5]
```diff
@@ -94,3 +94,66 @@ creates.
...
+             matches="some string")
+[{""}]
+
+>>> getPools(available=true,
+             matches="some string")
+[{""}]
+
+```
+
+* `getConsumers()`: Returns a list of consumers that a current entitlement offers.
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: Please remove getConsumers, we'll scope this in a different object/service.

File: docs/subscription-manager/dbus_objects.md [hunk 6]
```diff
@@ -94,3 +94,66 @@ creates.
...
+
+* `getProducts()`: Returns a list of products that a current entitlement offers.
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: Please remove getProducts, we'll scope this in a different object/service.

File: docs/subscription-manager/dbus_objects.md [hunk 7]
```diff
@@ -94,3 +94,66 @@ creates.
...
+* `getStatus()`: Returns a string describing a current entitlement status.
+
+   * "Future Subscription"
+   * "Subscribed"
+   * "Not Subscribed"
+   * "Expired"
+   * "Partially Subscribed"
+   * "Unknown"
+
+* `getSubscriptions(dictionary(string,variant))`: Returns a list of subscriptions that are available to the registered owner.
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: Amusingly, we don't actually expose the subscriptions at all in subscription-manager, instead we only expose the pools that are based on the subscriptions. Please remove this method.

File: docs/subscription-manager/dbus_objects.md [hunk 8]
```diff
@@ -94,3 +94,66 @@ creates.
...
+>>> getSubscriptions(consumed=true,
+                     matches="some string")
+[{""}]
+
+>>> getSubscriptions(available=true,
+                     matches="some string")
+[{""}]
+```
+
+* `getPools(dictionary(string,variant))`: Returns a list of pools that a current entitlement offers.
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: In order to stay consistent and reduce confusion, let's only use "entitlement" to refer to what you get when you attach (the certificate in /etc/pki/entitlement).

Accordingly, please reword this; my suggestion is "Returns a list of pools accessible by this system."

File: docs/subscription-manager/dbus_objects.md [hunk 9]
```diff
@@ -94,3 +94,66 @@ creates.
...
+Entitled environment includes:
+
+   * an owner of the entitled subscriptions
+   * a list of pools that the entitled subscriptions provide
+   * a list of products that the pools provide
+   * a list of subscriptions that the entitled environment provide
+   * a list of products that are installed in a system
+   
+## Methods
+* `getStatus()`: Returns a string describing a current entitlement status.
```
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: Please capitalize similar to the other d-bus method names: `GetStatus`

File: docs/subscription-manager/dbus_objects.md (hunk 8 above)
Comment (kahowell) [inline docs/subscription-manager/dbus_objects.md]: Also, please capitalize: `GetPools`

File: docs/subscription-manager/dbus_objects.md (hunk 1 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: fixed

File: docs/subscription-manager/dbus_objects.md (hunk 2 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: fixed

File: docs/subscription-manager/dbus_objects.md (hunk 3 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: fixed

File: docs/subscription-manager/dbus_objects.md (hunk 4 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: fixed

File: docs/subscription-manager/dbus_objects.md (hunk 9 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: done

File: docs/subscription-manager/dbus_objects.md (hunk 7 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: removed.

File: docs/subscription-manager/dbus_objects.md (hunk 5 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: removed

File: docs/subscription-manager/dbus_objects.md (hunk 6 above)
Comment (jstavel) [inline docs/subscription-manager/dbus_objects.md]: removed

**Comment by jstavel** [comment]
The code has been updated according to all comments.

**Comment by kahowell** [comment]
@jstavel looks good. Feel free to merge now if you'd like, or you can wait until you get the prototype implemented; Either is fine in my opinion.

**Comment by jstavel** [comment]
Hi @kahowell I have added examples of GetPools method usage. Mainly results. I have named the same way as 'subscription-manager list --available' does. The only thing I am not sure is 'service_type'. Whether it is array of strings, or string(). Big numbers I left as string.

**Comment by jstavel** [comment]
This is very old PR. I think there is nothing useful now.
